### ISSUE TITLE
App crashes when adding new share transaction

### ISSUE THREAD

**Issue by cmbarbu** [moneymanagerex/moneymanagerex#4778]
MMEX version:
 - [ ] 1.5.14
 - [ ] 1.5.15
 - [x] 1.5.16
 - [x] 1.5.17 Beta
 - [ ] Other (please specify)

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug

After adding a new share transaction, MMEX crashes. (On reopen, a dialog window states that the database is in an open state.)
The transaction seems to have been added successfully.

Tested with 1.5.16 as well as 1.5.17 Beta (145f583 (2022-07-03))

![mmex-bug-add-share-transaction](https://user-images.githubusercontent.com/1184162/177138780-8f059e29-9cf9-42d4-a303-d84d93826ed3.PNG)

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Select a stock portfolio
2. Add a new transaction

Expected result: The transaction is saved and appears in the transaction dialog.

Actual result: The application crashes.

**Comment by whalley** [comment]
I've tried this on a test database and I can't reproduce this. Is it possible to create a cut-down database that we can simulate the failure on.

**Comment by cmbarbu** [comment]
@whalley Sure. This is reproducible (for me, on Win64) also on a new database. Here is a link to a minimal DB: https://file.io/ZvdrVn2psNEB

More detailed steps to reproduce:
1. Select an Investment account
2. Select a share
3. Click on "Add Trans"
4. Fill in required fields (Share number & price, payee and category)
5. Click on "OK"

The result is MMEX closing without any message after the transaction is saved.

![share1](https://user-images.githubusercontent.com/1184162/177187996-6b11d353-86b1-4d1f-b466-130a0febeade.PNG)
![share2](https://user-images.githubusercontent.com/1184162/177188010-f758df98-29df-4b85-a11a-287213c2a82b.PNG)
![share3](https://user-images.githubusercontent.com/1184162/177188620-5f176b95-5142-424c-90bb-4b444c5b115c.PNG)

**Comment by renato-mmex** [comment]
probably:  https://github.com/moneymanagerex/moneymanagerex/issues/4412

**Comment by whalley** [comment]
OK, it does not fail on MacOS but I get the failure on Windows. As @renato-mmex notes it is likely related to https://github.com/moneymanagerex/moneymanagerex/issues/4412 .. I'll have to leave it to @vomikan or somebody else to debug this on Windows.

**Comment by vomikan** [comment]
@cmbarbu coul you share the test DB again?
I can't reproduce this issue.

**Comment by cmbarbu** [comment]
@vomikan Sure, here it is: https://file.io/edaDQmcGCEye

(Not sure why the link expires so quickly, it should be active for 2 weeks...)

**Comment by whalley** [comment]
@vomikan Without the 
`m_nav_tree_ctrl->SetFocus();`

I get MMEX starting like this (ignore the toolbar padding - that's another issue):
<img width="310" alt="CleanShot 2022-07-08 at 12 54 49@2x" src="https://user-images.githubusercontent.com/487345/177987369-51267912-818f-4bba-8535-0f56d452c9d2.png">

Which looks odd, I have to click on the nav tree ctrl to remove the toolbar icon highlighting.
<img width="310" alt="CleanShot 2022-07-08 at 12 52 04@2x" src="https://user-images.githubusercontent.com/487345/177986944-d9d1d7dc-6c75-4e2b-8529-18240c3c0469.png">

**Comment by vomikan** [comment]
This command crashes only after exiting the transaction dialog. I tried to figure out the reason, but I couldn't.

**Comment by vomikan** [comment]
On Win10 at startup
![image](https://user-images.githubusercontent.com/6836805/177999027-058991c8-f82c-451d-a9fb-ac661396d32b.png)

When transaction added
![image](https://user-images.githubusercontent.com/6836805/177999194-4d0c786c-39c7-4f72-b989-c88c0f72738c.png)

**Comment by whalley** [comment]
All good...
